packages/expo-image/ios/ImageUtils.swift

```diff
@@ -116,19 +116,42 @@ func shouldDownscale(image: UIImage, toSize size: CGSize, scale: Double) -> Bool
  Resizes the animated image to fit in the given size and scale.
  */
 func resize(animatedImage image: UIImage, toSize size: CGSize, scale: Double) async -> UIImage {
-  // For animated images, the `images` member is non-nil and represents an array of animation frames.
-  if let images = image.images {
-    // Resize each animation frame separately.
-    let resizedImages = await concurrentMap(images) { image in
+  if !image.sd_isAnimated {
+    return resize(image: image, toSize: size, scale: scale)
+  }
+
+  // Cast to SDAnimatedImage to access its properties
+  guard let image = image as? SDAnimatedImage else {
+    return resize(image: image, toSize: size, scale: scale)
+  }
+
+  let frameCount = await image.animatedImageFrameCount
```

Comment:
I guess we don't need `await` here 🤔 `animatedImageFrameCount` is just a member and doesn't take a closure as an argument
```suggestion
  let frameCount = image.animatedImageFrameCount
```

Reply:
@tsapeta That's what I thought as well at first, but if you check autocomplete:

![CleanShot 2024-04-21 at 18 51 22@2x](https://github.com/expo/expo/assets/19533289/5b95b547-3824-4e2d-846f-196d4348fc1f)

And you get a compiler warning:

![CleanShot 2024-04-21 at 18 55 23@2x](https://github.com/expo/expo/assets/19533289/5b200ca2-dfd9-4f83-bf65-2712ae5f96f8)


Swift apparently has async computed properties: https://www.hackingwithswift.com/quick-start/concurrency/how-to-create-and-use-async-properties and this is what is the case with these SDAnimatedImage properties. Same goes for image.animatedImageFrame

![CleanShot 2024-04-21 at 18 54 55@2x](https://github.com/expo/expo/assets/19533289/6a516787-c749-4848-8aab-c01d31778810)

Reply:
That's interesting. Looks like all Objective-C properties are always async within the async context 🤔 I would rather expect them to be always sync 😅 TIL